tests/integration/conftest.py

```diff
@@ -17,5 +19,40 @@ def vcr_config():
             "X-OpenAI-Client-User-Agent",
             "User-Agent",
         ],
-        "match_on": ["method", "uri", "body"],
+        "match_on": ["method", "body"],
     }
+
+
+def patch_api_base(requestor):
+    new_api_base = f"{PROXY}/v1"
+    requestor.api_base = new_api_base
+    return requestor
+
+
+@pytest.fixture
+def patched_api_requestor(mocker):
+    original_init = openai.api_requestor.APIRequestor.__init__
+    original_validate_headers = openai.api_requestor.APIRequestor._validate_headers
+
+    def patched_init(requestor, *args, **kwargs):
+        original_init(requestor, *args, **kwargs)
+        patch_api_base(requestor)
+
+    def patched_validate_headers(self, supplied_headers):
+        headers = original_validate_headers(self, supplied_headers)
+        headers["AGENT-MODE"] = os.environ.get(
+            "AGENT_MODE",
+            "AAAAAAAAAAAAAAAAAAAAAMLheAAaAAAA0%2BuSeid%2BULvsea4JtiGRiSDSJSI%3DEUifiRBkKG5E2XzMDjRfl76ZC9Ub0wnz4XsNiRVBChTYbJcE3F",
```

Comment:
What's with the default value checked in here?
It looks like a secret of some kind?